platform/ios/src/MGLScaleBarView.m

```diff
@@ -0,0 +1,387 @@
+#import <Mapbox/Mapbox.h>
+#import "MGLScaleBarView.h"
+
+
+@class MGLScaleBarLabel;
+
+@interface MGLScaleBarView()
+@property (nonatomic, assign, getter=isVisible) BOOL visible;
+@property (nonatomic) NSArray<MGLScaleBarLabel *> *labels;
+@property (nonatomic) NSArray<UIView *> *bars;
+@property (nonatomic) UIView *containerView;
+@property (nonatomic) MGLDistanceFormatter *formatter;
+@property (nonatomic) NSArray<NSArray<NSNumber *> *> *metricTable;
+@property (nonatomic) NSArray<NSArray<NSNumber *> *> *imperialTable;
+@property (nonatomic) NSArray<NSNumber*> *row;
+@property (nonatomic) UIColor *primaryColor;
+@property (nonatomic) UIColor *secondaryColor;
+@property (nonatomic) CALayer *borderLayer;
+@property (nonatomic, assign) CGFloat borderWidth;
+@end
+
+static const CGFloat MGLBarHeight = 4;
+static const CGFloat MGLFeetPerMeter = 3.28084;
+static const CGFloat MGLFeetPerMile = 5280;
+
+@interface MGLScaleBarLabel : UILabel
+@end
+
+@implementation MGLScaleBarLabel
+
+- (void)drawTextInRect:(CGRect)rect {
+    CGSize shadowOffset = self.shadowOffset;
+    UIColor *textColor = self.textColor;
+    
+    CGContextRef context = UIGraphicsGetCurrentContext();
+    CGContextSetLineWidth(context, 1);
```

Comment:
The stroke width here (around the outside of the label’s numbers/text) feels a bit too thin and diminishes legibility.

![stroke-light](https://cloud.githubusercontent.com/assets/1198851/24689716/6f2b1184-1996-11e7-8469-8a2eb87a96b1.png)
![dark-stroke](https://cloud.githubusercontent.com/assets/1198851/24689715/6f210f5e-1996-11e7-99cf-097d319ccf6a.png)
_Top: `1` unit; bottom: `2` units_

It does look like widening the stroke width does cause the sides to get cropped a bit, though... especially the zero character — its stroke is very uneven and sharp when `2`.